### PR TITLE
Review secret recovery

### PR DESCRIPTION
**Pull request by benoitdevos**
Provide one-stop API to Legal Officer for reviewing both account and secret recovery requests.

logion-network/logion-internal#1266

### CODE REVIEW

File: packages/client/src/RecoveryReviewClient.ts [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+            throw new Error(`Unsupported recovery type ${ type }`);
+        }
+    }
+
+    async acceptRecoveryRequest(params: RecoveryBackendParameters): Promise<void> {
+        const { type, id } = params;
+        if (type === "ACCOUNT") {
+            await this.backend().post(`/api/protection-request/${ id }/accept`);
+        } else {
+            await this.backend().post(`/api/secret-recovery/${ id }/accept`);
```
Comment (gdethier) [inline packages/client/src/RecoveryReviewClient.ts:79]: Should be included in a try-catch block and use `backendError` in catch.

File: packages/client/src/RecoveryReviewClient.ts [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        }
+    }
+
+    async rejectRecoveryRequest(params: RejectRecoveryBackendParameters): Promise<void> {
+        const { type, id, rejectReason } = params;
+        if (type === "ACCOUNT") {
+            await this.backend().post(`/api/protection-request/${ id }/reject`, {
```
Comment (gdethier) [inline packages/client/src/RecoveryReviewClient.ts:86]: Should be included in a try-catch block and use `backendError` in catch.

File: packages/client/src/RecoveryReviewClient.ts [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+                rejectReason,
+            });
+        } else {
+            await this.backend().post(`/api/secret-recovery/${ id }/reject`, {
```
Comment (gdethier) [inline packages/client/src/RecoveryReviewClient.ts:90]: Should be included in a try-catch block and use `backendError` in catch.

**Review by gdethier — APPROVED** [review]
Good job! I left a couple of suggestions.

File: packages/client/src/RecoveryReviewClient.ts (hunk 3 above)
Comment (benoitdevos) [reply packages/client/src/RecoveryReviewClient.ts]: Done.

File: packages/client/src/RecoveryReviewClient.ts (hunk 2 above)
Comment (benoitdevos) [reply packages/client/src/RecoveryReviewClient.ts]: Done.

File: packages/client/src/RecoveryReviewClient.ts (hunk 1 above)
Comment (benoitdevos) [reply packages/client/src/RecoveryReviewClient.ts]: Done.

File: packages/client/src/RecoveryReviewClient.ts [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+            return response.data;
+        } else {
+            throw new Error(`Unsupported recovery type ${ type }`);
+        }
+    }
+
+    async acceptRecoveryRequest(params: RecoveryBackendParameters): Promise<void> {
+        const { type, id } = params;
+        if (type === "ACCOUNT") {
+            await this.backend().post(`/api/protection-request/${ id }/accept`);
```
[the remark this answers is not in the document]
Comment (benoitdevos) [reply packages/client/src/RecoveryReviewClient.ts]: Done.

File: packages/client/src/RecoveryReviewClient.ts [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+        return response.data;
+    }
+
+    async fetchRecoveryInfo(params: RecoveryBackendParameters): Promise<RecoveryInfo> {
+        const { type, id } = params;
+        if (type === "ACCOUNT") {
+            const response = await this.backend().put(`/api/protection-request/${ id }/recovery-info`);
+            return response.data;
+        } else if (type === "SECRET") {
+            const response = await this.backend().put(`/api/secret-recovery/${ id }/recovery-info`);
```
[the remark this answers is not in the document]
Comment (benoitdevos) [reply packages/client/src/RecoveryReviewClient.ts]: Done.

File: packages/client/src/RecoveryReviewClient.ts [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+
+    async fetchRecoveryRequests(): Promise<BackendRecoveryRequest[]> {
+        const response = await this.backend().put(`/api/recovery-requests`);
+        return response.data;
+    }
+
+    async fetchRecoveryInfo(params: RecoveryBackendParameters): Promise<RecoveryInfo> {
+        const { type, id } = params;
+        if (type === "ACCOUNT") {
+            const response = await this.backend().put(`/api/protection-request/${ id }/recovery-info`);
```
[the remark this answers is not in the document]
Comment (benoitdevos) [reply packages/client/src/RecoveryReviewClient.ts]: Done.

File: packages/client/src/RecoveryReviewClient.ts [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+export class RecoveryReviewClient {
+
+    private readonly legalOfficer: LegalOfficerClass;
+
+    constructor(legalOfficer: LegalOfficerClass) {
+        this.legalOfficer = legalOfficer;
+    }
+
+    async fetchRecoveryRequests(): Promise<BackendRecoveryRequest[]> {
+        const response = await this.backend().put(`/api/recovery-requests`);
```
[the remark this answers is not in the document]
Comment (benoitdevos) [reply packages/client/src/RecoveryReviewClient.ts]: Done.

File: packages/client/src/RecoveryReview.ts [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+    constructor(data: BackendRecoveryRequest, client: RecoveryReviewClient) {
+        this.data = data;
+        this.client = client;
+    }
+
+    readonly data: BackendRecoveryRequest;
+    protected readonly client: RecoveryReviewClient;
+}
+
+export class NonPendingRecoveryRequest extends RecoveryRequest {
```
[the remark this answers is not in the document]
Comment (benoitdevos) [reply packages/client/src/RecoveryReview.ts]: Good suggestion, I was struggling on naming on this...

File: packages/client/src/RecoveryReview.ts [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+        return requests.filter(request => request.status === "PENDING")
+            .map(request => new PendingRecoveryRequest(request, this.client));
+    }
+
+    async fetchRecoveryRequestsHistory(): Promise<NonPendingRecoveryRequest[]> {
+        const requests = await this.client.fetchRecoveryRequests();
+        return requests.filter(request => request.status !== "PENDING")
+            .map(request => new NonPendingRecoveryRequest(request, this.client));
+    }
+}
```
[the remark this answers is not in the document]
Comment (benoitdevos) [reply packages/client/src/RecoveryReview.ts:85]: Good suggestion, done.

**Review by gdethier — APPROVED** [review]
:+1:

**Pull request merged**